### PR TITLE
KOGITO-4094 Avoid to deploy artifacts if author is not `kiegroup` and no artifacts repository has been given

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-4094

Related PRs:
- https://github.com/kiegroup/kogito-runtimes/pull/962
- https://github.com/kiegroup/optaplanner/pull/1095
- https://github.com/kiegroup/kogito-apps/pull/590
- https://github.com/kiegroup/kogito-examples/pull/515

### CODE REVIEW

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -394,6 +404,10 @@ String getBotAuthorCredsID(){
     return params.BOT_CREDENTIALS_ID
 }
 
+boolean isNotTestingBuild(){
```
Comment (rsynek) [inline Jenkinsfile.deploy:407]:
```suggestion
boolean isNotTestingBuild() {
```

File: Jenkinsfile.promote [hunk 2]
```diff
@@ -374,6 +374,10 @@ String getSnapshotBranch(){
     return "${getNextMicroSnapshotVersion(getProjectVersion()).toLowerCase()}-${env.BOT_BRANCH_HASH}"
 }
 
+boolean isNotTestingBuild(){
```
Comment (rsynek) [inline Jenkinsfile.promote:377]:
```suggestion
boolean isNotTestingBuild() {
```

File: Jenkinsfile.promote [hunk 3]
```diff
@@ -464,9 +468,11 @@ MavenCommand getMavenCommand() {
 }
 
 void runMavenDeploy(MavenCommand mvnCmd) {
-    mvnCmd = mvnCmd.clone()
-    if(params.MAVEN_DEPLOY_REPOSITORY){
-        mvnCmd.withDeployRepository(params.MAVEN_DEPLOY_REPOSITORY)
+    if(isNotTestingBuild() || params.MAVEN_DEPLOY_REPOSITORY){
```
Comment (rsynek) [inline Jenkinsfile.promote:471]:
```suggestion
    if(isNotTestingBuild() || params.MAVEN_DEPLOY_REPOSITORY) {
```

**Review by rsynek — COMMENTED** [review]
Formatting suggestions, otherwise LGTM.

**Comment by radtriste** [comment]
tested all paths ok

**Comment by radtriste** [comment]
@rsynek was the `Linux - Pull Request` set back on purpose for `master` branch ?

**Comment by rsynek** [comment]
@radtriste it should have been already replaced by "(build) optaplanner", which should also take care of the SonarCloud analysis, right?

**Comment by radtriste** [comment]
yep. I will PR `kie-jenkins-script` to avoid this job to exist ;)

File: Jenkinsfile.deploy [hunk 4]
```diff
@@ -171,12 +171,53 @@ pipeline {
...
+                script {
+                    runMavenDeploy(getOptaplannerMavenCommand(), params.SKIP_TESTS)
+                    runMavenDeploy(getOptawebVehicleRoutingMavenCommand(), params.SKIP_TESTS)
+                    runMavenDeploy(getOptawebEmployeeRosteringMavenCommand(), params.SKIP_TESTS)
+                }
+            }
+        }
+        stage('Deploy locally') {
+            when {
+                expression { return shouldStageArtifacts() || isSpecificArtifactsUpload() }
```
Comment (rsynek) [inline Jenkinsfile.deploy:188]: When `shouldStageArtifacts()` is `true`, both this stage and the `Stage artifacts` will run. Is that intentional?

Comment (radtriste) [reply Jenkinsfile.deploy:188]: yes, because we stage all artifacts which were deployed in local folder so we need this :)

**Comment by radtriste** [comment]
> ill PR `kie-jenkins-script` to avoid this job to exist ;)

@rsynek see https://github.com/kiegroup/kie-jenkins-scripts/pull/1026 from Michael

**Comment by mbiarnes** [comment]
**Linux - Pull Request** build will not be executed any more for optaplanner PRs into master branch.
https://github.com/kiegroup/kie-jenkins-scripts/pull/1026

**Review by rsynek — APPROVED** [review]

**Pull request merged**